src/Uno.UI/UI/Xaml/DragView.cs

```diff
@@ -0,0 +1,192 @@
+
+		public ImageSource? Content
+		{
+			get => (ImageSource?)GetValue(ContentProperty);
+			private set => SetValue(ContentProperty, value);
+		}
+		#endregion
+
+		#region ContentVisibility
+		public static readonly DependencyProperty ContentVisibilityProperty = DependencyProperty.Register(
+			"ContentVisibility", typeof(Visibility), typeof(DragView), new PropertyMetadata(default(Visibility)));
+
+		public Visibility ContentVisibility
+		{
+			get => (Visibility)GetValue(ContentVisibilityProperty);
+			private set => SetValue(ContentVisibilityProperty, value);
+		}
+		#endregion
+
+		#region TooltipVisibility
+		public static readonly DependencyProperty TooltipVisibilityProperty = DependencyProperty.Register(
+			"TooltipVisibility", typeof(Visibility), typeof(DragView), new PropertyMetadata(default(Visibility)));
+
+		public Visibility TooltipVisibility
+		{
+			get => (Visibility)GetValue(TooltipVisibilityProperty);
+			set => SetValue(TooltipVisibilityProperty, value);
+		} 
+		#endregion
+
+		private readonly DragUI? _ui;
+		private readonly TranslateTransform _transform;
+
+		private Point _location;
+
+		public DragView(DragUI? ui)
+		{
+			_ui = ui;
+			DefaultStyleKey = typeof(DragView);
+			RenderTransform = _transform = new TranslateTransform();
+
+			Content = ui?.Content;
+		}
+
+		public void SetLocation(Point location)
+		{
+			// TODO: Make sure to not move the element out of the bounds of the window
+			_location = location;
+
+			_transform.X = location.X - (ActualWidth / 2);
+			_transform.Y = location.Y - 40; // The caption is above the pointer
+		}
+
+		public void Update(DataPackageOperation acceptedOperation, CoreDragUIOverride viewOverride)
+		{
+			// UWP does not allow new lines (trim to the first line, even if blank) and trims the text.
+			var caption = viewOverride
+				.Caption
+				?.Split(new[] {'\r', '\n'}, StringSplitOptions.None)
+				.FirstOrDefault()
+				?.Trim();
+
+			if (string.IsNullOrEmpty(caption))
+			{
+				caption = ToCaption(acceptedOperation);
+			}
+
+			Glyph = ToGlyph(acceptedOperation);
+			GlyphVisibility = ToVisibility(viewOverride.IsGlyphVisible);
+			Caption = caption!;
+			CaptionVisibility = ToVisibility(viewOverride.IsCaptionVisible && !string.IsNullOrWhiteSpace(caption));
+			Content = viewOverride.Content as ImageSource ?? _ui?.Content;
+			ContentVisibility = ToVisibility(viewOverride.IsContentVisible);
+			TooltipVisibility = ToVisibility(viewOverride.IsGlyphVisible || viewOverride.IsCaptionVisible);
+			Visibility = Visibility.Visible;
+		}
+
+		public void Hide()
+		{
+			Visibility = Visibility.Collapsed;
+		}
+
+		private static Visibility ToVisibility(bool isVisible)
+			=> isVisible ? Visibility.Visible : Visibility.Collapsed;
+
+		private static string ToGlyph(DataPackageOperation result)
+		{
+			// If multiple flags set (which should not!), the UWP precedence is Link > Copy > Move
+			// TODO: Real glyph
```

Comment:
This TODO and the one below warrant issues to track them

Reply:
Good point, I'll try to fix it before the merge, otherwise I'll create a new PR for this one